tasks/ThrusterTask.cpp

```diff
@@ -98,13 +99,13 @@ void ThrusterTask::cleanupHook()
 
 void ThrusterTask::setGazeboModel( std::string const& pluginName, ModelPtr model )
 {
-    string worldName = GzGet((*(model->GetWorld())), Name, ());
-
-    string taskName = std::regex_replace(pluginName, std::regex("__"), "::");
-    provides()->setName(taskName);
-    _name.set(taskName);
+    topicName = std::regex_replace(pluginName, std::regex("__"), "/") + "/thrusters";
+}
 
-    string topicName = std::regex_replace(pluginName, std::regex("__"), "/") + "/thrusters";
+void ThrusterTask::setGazeboPluginTaskName( std::string const& pluginTaskName )
+{
+    provides()->setName(pluginTaskName);
+    _name.set(pluginTaskName);
 }
```

Comment:
Since everyone has the same exact implementation of this method, you could implement it in `ModelPluginTask`